File: jsonexport/Scripts/Editor/Exporter/JsonTreeInstance.cs
using UnityEditor;
using UnityEngine;
using System.Collections.Generic;

namespace SceneExport{
	[System.Serializable]
	public class JsonTreeInstance: IFastJsonValue{
		public TreeInstance data;
		public void writeRawJsonValue(FastJsonWriter writer){
			writer.beginRawObject();
			writer.writeKeyVal("color", data.color);
			writer.writeKeyVal("heightScale", data.heightScale);
			writer.writeKeyVal("lightmapColor", data.lightmapColor);
			writer.writeKeyVal("position", data.position);
			writer.writeKeyVal("rotation", data.rotation);//this is an angle and not a quaternion, surprisingly
			writer.writeKeyVal("prototypeIndex", data.prototypeIndex);
			writer.writeKeyVal("widthScale", data.widthScale);			
			writer.endObject();
		}
		
		public JsonTreeInstance(TreeInstance inst){
			data = inst;
		}
	}
}

File: ExodusImport/Source/ExodusImport/Private/JsonObjects/JsonTreeInstance.cpp
#include "JsonImportPrivatePCH.h"
#include "JsonTreeInstance.h"
#include "macros.h"

//#define JSON_ENABLE_VALUE_LOGGING

JsonTreeInstance::JsonTreeInstance(JsonObjPtr data){
	load(data);
}

void JsonTreeInstance::load(JsonObjPtr data){
	JSON_GET_VAR(data, color);
	JSON_GET_VAR(data, heightScale);
	JSON_GET_VAR(data, lightmapColor);
	JSON_GET_VAR(data, position);
	JSON_GET_VAR(data, rotation);
	JSON_GET_VAR(data, prototypeIndex);
	JSON_GET_VAR(data, widthScale);
}
